homeassistant/components/tado/climate.py

```diff
@@ -128,16 +144,23 @@ class TadoClimate(ClimateDevice):
 
     def __init__(
         self,
+        hass,
         tado,
         zone_name,
         zone_id,
         zone_type,
-        min_temp,
-        max_temp,
-        step,
-        ac_support_heat,
+        heat_min_temp,
+        heat_max_temp,
+        heat_step,
+        cool_min_temp,
+        cool_max_temp,
+        cool_step,
+        supported_hvac_modes,
+        supported_fan_modes,
+        support_flags,
     ):
         """Initialize of Tado climate entity."""
+        self.hass = hass
```

Comment:
No need to pass in `hass`. That is being set when passed to `async_add_entities`.

Reply:
If you need this because you added a call to `self._async_update_zone_data()`, then call that inside `async_added_to_hass` callback.

Reply:
I'm pretty sure I don't need this anymore and I refactored out the original reason why I added it.  I'm double-checking though

Reply:
All still looks good with it removed.